File: ShoeStore/ShoeStore.Domain/Entities/Order.cs
namespace ShoeStore.Domain.Entities;

public class Order
{
    public long Id { get; set; }
    public string OrderNumber { get; set; } = null!;
    public long CustomerId { get; set; }
    public long? CreatedBy { get; set; }
    public int? StoreId { get; set; }
    public int StatusId { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderType OrderType { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public User Customer { get; set; } = null!;
    public User? Creator { get; set; }
    public Store? Store { get; set; }
    public Status Status { get; set; } = null!;
    public ICollection<OrderDetail>? OrderDetails { get; set; }
}


public enum OrderType
{
    Online,
    Offline
}

public enum PaymentMethod
{
    Cash,
    Transfer
}

File: shoestorefe/lib/domain/entities/order.dart
import 'order_detail.dart';

class Order {
  final int id;
  final String orderNumber;
  final int customerId;
  final String? customerName;
  final int? createdBy;
  final String? creatorName;
  final int? storeId;
  final String? storeName;
  final int statusId;
  final double totalAmount;
  final int orderType;
  final int paymentMethod;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final List<OrderDetail> details;

  Order({
    required this.id,
    required this.orderNumber,
    required this.customerId,
    this.customerName,
    this.createdBy,
    this.creatorName,
    this.storeId,
    this.storeName,
    required this.statusId,
    required this.totalAmount,
    required this.orderType,
    required this.paymentMethod,
    required this.createdAt,
    this.updatedAt,
    required this.details,
  });
}
